core/account-id/src/lib.rs

```diff
@@ -298,42 +238,35 @@ impl AccountId {
     /// use near_account_id::AccountId;
     ///
     /// let alice = AccountId::new_unvalidated("alice.near".to_string());
-    /// assert!(AccountId::validate(alice.as_ref()).is_ok());
+    /// assert!(AccountId::validate(alice.as_str()).is_ok());
     ///
     /// let ƒelicia = AccountId::new_unvalidated("ƒelicia.near".to_string());
-    /// assert!(AccountId::validate(ƒelicia.as_ref()).is_err());
+    /// assert!(AccountId::validate(ƒelicia.as_str()).is_err());
     /// ```
     #[cfg(feature = "internal_unstable")]
     #[deprecated(since = "#4440", note = "AccountId construction without validation is illegal")]
     pub fn new_unvalidated(account_id: String) -> Self {
         Self(account_id.into())
     }
+}
 
-    pub fn is_system(&self) -> bool {
-        self.as_ref() == "system"
-    }
-
-    pub fn system_account() -> Self {
-        "system".parse().unwrap()
-    }
+impl std::ops::Deref for AccountId {
```

Comment:
optional: i like to `impl ops::Trait for X` for all `ops` trait. That's usually the neatest way, as sometimes you impl more than one trait from there.

Reply:
Yeah, only, in this case, with one use, I'm not convinced that it is as useful.